contrib/Dockerfile.test

```diff
@@ -11,72 +11,89 @@
 # docker build -t containerd-test --build-arg RUNC_VERSION=v1.0.0-rc94 -f Dockerfile.test ../
 
 ARG GOLANG_VERSION=1.16.6
+ARG GOLANG_IMAGE=golang
```

Comment:
Seems switching to GitHub Packages doesn't solve #5748 fully since our Linux in macOS tests use not just golang but also nginx and possibly others. So I'm fine removing GOLANG_IMAGE variable.

Reply:
I'm ok to keep it here; it has a default value set, so doesn't _require_ a `--build-arg` to be passed, but it does keep the option open to use it if needed.